### ISSUE TITLE
User mention dosen't work with E2E

### ISSUE THREAD

**Issue by arthur-trt** [RocketChat/Rocket.Chat#29645]
### Description:

When E2E is enable in Channel, Discussion or Private Message mention to user didn't work, only `@here` and `@all` work. 

### Steps to reproduce:

1. Enable E2E
2. Mention user

### Expected behavior:

Mention user should work on E2E channel

### Actual behavior:

![Screenshot 2023-06-26 at 10 46 12](https://github.com/RocketChat/Rocket.Chat/assets/4441676/a580f18e-c082-4d3d-9405-6d825b8f04ad)

### Server Setup Information:

- Version of Rocket.Chat Server: 6.2.8
- Operating System: Linux / K8s
- Deployment Method: Helm
- Number of Running Instances: 1 
- DB Replicaset Oplog:  5.0.16 / wiredTiger (oplog Activé)
- NodeJS Version:  14.21.3
- MongoDB Version: 5.0.16

### Client Setup Information

- Desktop App or Browser Version: Desktop 3.9.5 / Same on browser with Chrome last version
- Operating System: MacOs

### Relevant logs:

**Request with E2E :**
URL `POST https://rocket.[REDACTED]/api/v1/method.call/sendMessage`
Payload : 
```json
{"msg":"method","id":"191","method":"sendMessage","params":[{"_id":"6eWhrirMzKiw9oLGH","rid":"TAdfQaaLomiepuRmT","msg":"eyJhbGciOiJBIR1o+HFiDrN3FRRs4WU00WQclNPDv4jKi+W+eOMW6dgdtgLdmKBuqsD242tHcSE79D21hk/dGaNUB10HTv9SlnwIuqtGte0kDp1KTKF39suoDz5ENmyZJGnoPqaWMqPyc3ygic6iw/j3t/XEz4lqrQUW2att2tfU/d8YzWBmkaM=","t":"e2e","e2e":"pending"}]}
```
Response: 
```json
  {
    "message": "{\"msg\":\"result\",\"id\":\"191\",\"result\":{\"_id\":\"6eWhrirMzKiw9oLGH\",\"rid\":\"TAdfQaaLomiepuRmT\",\"msg\":\"eyJhbGciOiJBIR1o+HFiDrN3FRRs4WU00WQclNPDv4jKi+W+eOMW6dgdtgLdmKBuqsD242tHcSE79D21hk/dGaNUB10HTv9SlnwIuqtGte0kDp1KTKF39suoDz5ENmyZJGnoPqaWMqPyc3ygic6iw/j3t/XEz4lqrQUW2att2tfU/d8YzWBmkaM=\",\"t\":\"e2e\",\"e2e\":\"pending\",\"ts\":{\"$date\":1687769378943},\"u\":{\"_id\":\"NPZtAHD8RwfarpprT\",\"username\":\"Arthur\",\"name\":\"Arthur [REDACTED]\"},\"_updatedAt\":{\"$date\":1687769379141},\"urls\":[],\"mentions\":[],\"channels\":[]}}",
    "success": true
}
```

**Request without E2E:**
URL: `POST https://rocket.[REDACTED]/api/v1/method.call/sendMessage`
Payload: 
```json
  {"msg":"method","id":"204","method":"sendMessage","params":[{"_id":"GXSoHPpWJw7E6ehw2","rid":"TAdfQaaLomiepuRmT","msg":"@Lucas this is a test"}]}
 ```
 Response: 
```json
{
    "message": "{\"msg\":\"result\",\"id\":\"204\",\"result\":{\"_id\":\"GXSoHPpWJw7E6ehw2\",\"rid\":\"TAdfQaaLomiepuRmT\",\"msg\":\"@Lucas this is a test\",\"ts\":{\"$date\":1687769593643},\"u\":{\"_id\":\"NPZtAHD8RwfarpprT\",\"username\":\"Arthur\",\"name\":\"Arthur [REDACTED]\"},\"_updatedAt\":{\"$date\":1687769593803},\"urls\":[],\"mentions\":[{\"_id\":\"6zC2RCik7Rbp2n4ou\",\"name\":\"Lucas [REDACTED]\",\"username\":\"Lucas\",\"type\":\"user\"}],\"channels\":[],\"md\":[{\"type\":\"PARAGRAPH\",\"value\":[{\"type\":\"MENTION_USER\",\"value\":{\"type\":\"PLAIN_TEXT\",\"value\":\"Lucas\"}},{\"type\":\"PLAIN_TEXT\",\"value\":\" this is a test\"}]}]}}",
    "success": true
}
```

**Comment by arthur-trt** [comment]
Still present in 6.5.0

**Comment by milton-rucks** [comment]
Hi @arthur-trt I could not reproduce it on my end.
Is it possible to check if the problem persists in 6.6.x?

**Comment by arthur-trt** [comment]
The problem seems to persist in 6.6.3

**Comment by arthur-trt** [comment]
Still present in 6.7.0

![Screenshot 2024-04-09 at 09 28 51](https://github.com/RocketChat/Rocket.Chat/assets/4441676/d8860992-9e02-4cfd-9887-6dcde83e33fa)

**Without E2E:** 
Payload :
```json
{"msg":"method","id":"221","method":"sendMessage","params":[{"_id":"srvbghmmc5FqaW4TA","rid":"TAdfQaaLomiepuRmT","msg":"@Romain test for issue"},null]}
```
Response:
```json
{
  "message": "{\"msg\":\"result\",\"id\":\"221\",\"result\":{\"_id\":\"srvbghmmc5FqaW4TA\",\"rid\":\"TAdfQaaLomiepuRmT\",\"msg\":\"@Romain test for issue\",\"ts\":{\"$date\":1712647660816},\"u\":{\"_id\":\"NPZtAHD8RwfarpprT\",\"username\":\"Arthur\",\"name\":\"Arthur [REDACTED]\"},\"_updatedAt\":{\"$date\":1712647660853},\"urls\":[],\"mentions\":[{\"_id\":\"bcFQJdbR5R8xCCXwg\",\"name\":\"Romain [REDACTED]\",\"username\":\"Romain\",\"type\":\"user\"}],\"channels\":[],\"md\":[{\"type\":\"PARAGRAPH\",\"value\":[{\"type\":\"MENTION_USER\",\"value\":{\"type\":\"PLAIN_TEXT\",\"value\":\"Romain\"}},{\"type\":\"PLAIN_TEXT\",\"value\":\" test for issue\"}]}]}}",
  "success": true
}
```

**With E2E:**
Payload: 
```json
{"msg":"method","id":"230","method":"sendMessage","params":[{"_id":"uACDcr2GnALuYN8LA","rid":"TAdfQaaLomiepuRmT","msg":"eyJhbGciOiJBQJyNTMxZv+sa2b9IGSCAPoRvDCpz+cwD5jPeBKpoUjmxDu5JvG4JwTLa11W4Mo7MlHpijRFbyG0ntP+dNBMk4NSsOKjJphf3kuB8vduDIU4phcwspYyc7Dvgap4SY87Y5/kIDRgqn37JzHihMS1/XwZ2UyvB4x90RwDHgm+L+x/NVkuO656wbzDel2dDHVmr","t":"e2e","e2e":"pending"},null]}
```

Response: 
```json
{
  "message": "{\"msg\":\"result\",\"id\":\"230\",\"result\":{\"_id\":\"uACDcr2GnALuYN8LA\",\"rid\":\"TAdfQaaLomiepuRmT\",\"msg\":\"eyJhbGciOiJBQJyNTMxZv+sa2b9IGSCAPoRvDCpz+cwD5jPeBKpoUjmxDu5JvG4JwTLa11W4Mo7MlHpijRFbyG0ntP+dNBMk4NSsOKjJphf3kuB8vduDIU4phcwspYyc7Dvgap4SY87Y5/kIDRgqn37JzHihMS1/XwZ2UyvB4x90RwDHgm+L+x/NVkuO656wbzDel2dDHVmr\",\"t\":\"e2e\",\"e2e\":\"pending\",\"ts\":{\"$date\":1712647671261},\"u\":{\"_id\":\"NPZtAHD8RwfarpprT\",\"username\":\"Arthur\",\"name\":\"Arthur [REDACTED]\"},\"_updatedAt\":{\"$date\":1712647671283},\"urls\":[],\"mentions\":[],\"channels\":[]}}",
  "success": true
}
```

**Details about current version:**

Version : 6.7.0
Apps Engine Version : 1.42.1
Node Version : v14.21.3
Database Migration : 304 (April 9, 2024 9:25)
MongoDB : 5.0.23 / wiredTiger (oplog Disabled)
Release 6.7.0

Present on :
Desktop MacOS M1 3.9.14

**Comment by arthur-trt** [comment]
Fixed in 7.0.0! :D
